types/block_test.go

```diff
@@ -44,6 +44,7 @@ func TestBlockValidateBasic(t *testing.T) {
 
 	block := MakeBlock(h, txs, commit, evList)
 	require.NotNil(t, block)
+	block.ProposerAddress = valSet.GetProposer().Address
```

Comment:
I think if we're going to require the proposer in ValidateBasic, we need to accommodate it in MakeBlock. Any block without it will fail validation.

Maybe the right middle ground is to change ValidateBasic to allow an empty proposer, but if its given it must be tmhash.Size